### PR TITLE
Add some missing Float cases

### PR DESCRIPTION
**Pull request by sparkprime**
I ran into some errors when removing the TODO from #588 
re: resolution of #594 

These are all the missing cases I could find, and the TODO in #588 can now be removed.

### CODE REVIEW

**Comment by mitchellh** [comment]
Thanks! Would you mind adding tests? If not, then we can add them, but it'll take a bit longer to merge.

**Comment by sparkprime** [comment]
I'm adding to  resource_data_test.go and resource_test.go, any more?

**Comment by sparkprime** [comment]
Actually I think resource_data_test is enough.

**Comment by sparkprime** [comment]
Ping

File: helper/schema/schema.go [hunk 1]
```diff
@@ -1084,6 +1087,12 @@ func (m schemaMap) validatePrimitive(
 		if err := mapstructure.WeakDecode(raw, &n); err != nil {
 			return nil, []error{err}
 		}
+	case TypeFloat:
+		// Verify that we can parse this as an int
```
Comment (mitchellh) [inline helper/schema/schema.go]: typo! doesn't matter though :) can fix easily later

**Comment by mitchellh** [comment]
Great! LGTM. Thanks.

**Pull request merged**